lib/rubygems/command.rb

```diff
@@ -456,7 +456,7 @@ def add_extra_args(args)
     until extra.empty? do
       ex = []
       ex << extra.shift
-      ex << extra.shift if extra.first.to_s =~ /^[^-]/
+      ex << extra.shift unless extra.first.to_s.start_with?("-")
```

Comment:
This changed the behavior! Apologies.

Reply:
This _would_ do the right thing, but at the cost of looking sad:

```ruby
      if (first_character = extra.first.to_s && first_character && !first_character.empty? && first_character.start_with?("-"))
        ex << extra.shift
      end
```

Reply:
Let's stick with `extra.shift if extra.first.to_s =~ /^[^-]/` for this then 😁

Reply:
@bronzdoc The sad was reduced (a little bit) in the above (I edited it in-place)..